plugins/out_es/es_conf.c

```diff
@@ -24,10 +24,93 @@
 #include <fluent-bit/flb_http_client.h>
 #include <fluent-bit/flb_signv4.h>
 #include <fluent-bit/flb_aws_credentials.h>
+#include <mbedtls/base64.h>
 
 #include "es.h"
 #include "es_conf.h"
 
+/*
+ extract_cloud_host gets a cloud_id encoded as base64
+ and extracts the cloud host.
+ Example:
+    input = ZXVyb3BxxxxxxZTA1Ng==
+    output = ba1XXXXXXXpl85114.europe-west1.gcp.cloud.es.io
+*/
+char * extract_cloud_host(struct flb_elasticsearch *ctx, char *cloud_id) {
+   ssize_t ret;
+   size_t olen = 180;
+   int items = 0;
+   char *host_hash = NULL;
+   char *host_region = NULL;
+
+   unsigned char dec_cloud_id[180] = { 0 };
+   struct mk_list *cloud_id_toks;
+   struct mk_list *cloud_host_toks;
+   struct mk_list *head;
+   struct flb_split_entry *entry;
+   struct mk_list *host_head;
+   /* keep only part after first ":" */
+   cloud_id_toks = flb_utils_split(cloud_id, ':', 1);
+   mk_list_foreach(head, cloud_id_toks) {
+       entry = mk_list_entry(head, struct flb_split_entry, _head);
+       cloud_id = entry->value;
+   }
+   /* decode base64 */
+   ret = mbedtls_base64_decode(dec_cloud_id, sizeof(dec_cloud_id), &olen,
+                            (unsigned char *) cloud_id, strlen(cloud_id));
+   if (ret != 0) {
+       flb_plg_error(ctx->ins, "cannot decode cloud_id");
+       return NULL;
+    }
+     /* extract strings */
+    cloud_host_toks = flb_utils_split((const char *)dec_cloud_id, '$', -1);
+    mk_list_foreach(host_head, cloud_host_toks) {
+        items++;
+        entry = mk_list_entry(host_head, struct flb_split_entry, _head);
+        if (items == 2) {
+          host_hash=entry->value;
+          break;
+        }
+        if (items == 1) {
```

Comment:
nit (this is my OCD talking 😅): can we put the conditionals in order? Like first evaluate the first token, then the second, then break.